revolsys-swing/src/main/java/com/revolsys/swing/map/layer/record/table/model/RecordLayerTableModel.java

```diff
@@ -195,13 +195,24 @@ public void forEachColumnDisplayValue(final Cancellable cancellable, final int c
 
   public void forEachColumnValue(final Cancellable cancellable, final int columnIndex,
     final Consumer<Object> action) {
-    final FieldDefinition field = getColumnFieldDefinition(columnIndex);
+    // final FieldDefinition field = getColumnFieldDefinition(columnIndex);
     final String fieldName = getColumnFieldName(columnIndex);
     final TableRecordsMode tableRecordsMode = getTableRecordsMode();
     if (tableRecordsMode != null && fieldName != null) {
       final Query query = getFilterQuery().clone();
-      addIdFieldNames(query);
-      query.select(field);
+
+      // EG: issue 492 (copy values from the TL table in the UI)
+      // What I think was happening here is this was trying to load only the
+      // tl_id and fieldname
+      // but the underlying code wants to load the entire TL record into the
+      // layer and cache it
+      // but it couldn't do this properly because the query told it to only load
+      // two attributes.
+      // The simplest fix appears to be to just tell it to load everything and
+      // not be smart.
+      //
+      // addIdFieldNames(query);
+      // query.select(field);
```

Comment:
This does seem a bit heavyweight to select eg. the entire table in order to just copy a list of values from one column? But I guess you've witnessed the performance and it is acceptable? One day this might need to be addressed again ie. for 6.x with a web api instead of direct db queries.

Reply:
I believe the 4.x stuff loaded the entire table and nobody complained about performance with that one.  Generally I hope users are only displaying what's in the their viewport in the table but when I tested it on all records a progress monitor pops up and shows you the progress so from a UI (With the cancel options that now works), so from a user experience point of view it seems ok.
You can't just query the database because it includes records that are modified/added/updated so changing the code to just query the database and bypass all the other stuff doesn't work.  And trying to change the other stuff to deal with cases where record definitions are different just didn't work.  As mentioned, think it tries to cache and reuse the records it loads.